### PR TITLE
Check data schema

### PR DESCRIPTION
**Pull request by neeleshdodda44**
Major changes:
- Dataset Manager checks raw data so that each dataset has a valid CSV format and that it has a header (assume column names are string). Sidenote: if data is all strings, how can we detect whether it has a header?
- Tests that ensure validation fails when DSM is given raw data without a header or raw data in an invalid CSV format.
- Cleaned up some code in DatasetManager

### CODE REVIEW

File: core/dataset_manager.py [hunk 1]
```diff
@@ -82,8 +82,48 @@ def __init__(self, config_manager):
...
+            files = os.listdir(folder_path)
+            for file in files:
+                if not file.endswith(".csv"): continue
+                if file[:2] == 'md': continue
+                file_path = os.path.join(folder_path, file)
+                try:
+                    dataset = pd.read_csv(file_path, index_col=False)
+                except Exception as e:
+                    logging.error(str(e))
+                    assert False, format_message.format(
```
Comment (georgymh) [inline core/dataset_manager.py]: Can we raise an exception instead?

File: tests/test_dataset_manager.py [hunk 2]
```diff
@@ -88,6 +104,39 @@ def drop_duplicates(df):
...
 
+def test_no_header(bad_config_manager_header):
+    """
+    Test that DSM invalidates dataset with no header.
+    """
+    error_message = ("No header has been provided in file {file} in folder "
+                     "{folder}")
+    try:
+        dsm = DatasetManager(bad_config_manager_header)
+        assert False, "Error should have been thrown for lack of header"
```
Comment (georgymh) [inline tests/test_dataset_manager.py]: Can you raise an exception here too?

File: tests/test_dataset_manager.py [hunk 3]
```diff
@@ -88,6 +104,39 @@ def drop_duplicates(df):
...
+    except AssertionError as e:
+        assert str(e) == error_message.format(
```
Comment (georgymh) [inline tests/test_dataset_manager.py]: Should this assert raise a meaningful error if the condition fails?

File: tests/test_dataset_manager.py [hunk 4]
```diff
@@ -88,6 +104,39 @@ def drop_duplicates(df):
...
+    """
+    Test that DSM invalidates dataset with invalid CSV format.
+    """
+    format_message = ("The file {file} in folder {folder} was improperly "
+                      "formatted. Please refer to the following error "
+                      "message from pandas for more information: {message}")
+                      
+    try:
+        dsm = DatasetManager(bad_config_manager_format)
+        assert False, "Error should have been thrown for bad format"
```
Comment (georgymh) [inline tests/test_dataset_manager.py]: Same as above.

File: tests/test_dataset_manager.py [hunk 5]
```diff
@@ -88,6 +104,39 @@ def drop_duplicates(df):
...
+    except AssertionError as e:
+        assert str(e) == format_message.format(
```
Comment (georgymh) [inline tests/test_dataset_manager.py]: Same as above.

File: core/dataset_manager.py (hunk 1 above)
Comment (neeleshdodda44) [reply core/dataset_manager.py]: Sure.

File: tests/test_dataset_manager.py (hunk 2 above)
Comment (neeleshdodda44) [reply tests/test_dataset_manager.py]: Sure.

File: tests/test_dataset_manager.py (hunk 3 above)
Comment (neeleshdodda44) [reply tests/test_dataset_manager.py]: Yeah, will add.

File: tests/test_dataset_manager.py (hunk 4 above)
Comment (neeleshdodda44) [reply tests/test_dataset_manager.py]: Sure.

File: tests/test_dataset_manager.py (hunk 5 above)
Comment (neeleshdodda44) [reply tests/test_dataset_manager.py]: Sure.

**Comment by neeleshdodda44** [comment]
Pretty much all of the algorithms I've seen that detect headers basically check if the first line has comma delimited strings. So tl;dr, I'm not aware of a fix as of now, but we can worry about this later.

Maybe this could be designated as an issue for the future?

**Pull request merged**